### PR TITLE
warn if version is not the latest

### PR DESCRIPTION
**Pull request by yann300**

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -14,7 +16,26 @@ registerPackageProtocol(cacheDir)
 
 const remixIdeUrl = 'package://6fd22d6fe5549ad4c4d8fd3ca0b7816b.mod'
 
-console.log('running', version)
+async function warnLatestVersion (current) {
+  try {
+    const res = await fetch('https://api.github.com/repos/ethereum/remix-desktop/releases/latest')
```
Comment (Aniket-Engg) [inline main.js:21]: I am not sure if `remix-desktop` tag should be checked as this implementation is for `remixd`

File: main.js [hunk 2]
```diff
@@ -14,7 +16,26 @@ registerPackageProtocol(cacheDir)
...
+    let latest = (await res.json()).tag_name
+    latest = latest.indexOf('v') === 0 ? latest.replace('v', '') : latest
+    if (semver.eq(latest, current)) {
+      console.log('\x1b[32m%s\x1b[0m', `[INFO] you are using the latest version ${latest}`)
+    } else if (semver.gt(latest, current)) {
+      console.log('\x1b[33m%s\x1b[0m', `[WARN] latest update is ${latest}, you are using ${current}`)
```
Comment (Aniket-Engg) [inline main.js:27]: May be `latest update` can be replaced with `latest version of remixd`

File: main.js [hunk 3]
```diff
@@ -14,7 +16,26 @@ registerPackageProtocol(cacheDir)
...
+      console.log('\x1b[33m%s\x1b[0m', `[WARN] please update using the following command:`)
+      console.log('\x1b[33m%s\x1b[0m', `[WARN] npm install @remix-project/remixd -g`)
+    }
+  } catch (e) {
+    console.log('unable to verify latest version')
+    console.log(e)
+  }  
+}
+
+warnLatestVersion(version)
```
Comment (Aniket-Engg) [inline main.js:37]: `await` missing

File: package.json [hunk 4]
```diff
@@ -71,9 +71,12 @@
     }
   },
   "dependencies": {
+    "@remix-project/remixd": "^0.3.2",
```
Comment (Aniket-Engg) [inline package.json:74]: latest version of remixd is `0.3.5`

File: main.js (hunk 1 above)
Comment (yann300) [reply main.js:21]: here we are checking for new update of remix-desktop, not remixd

File: main.js [hunk 5]
```diff
@@ -78,12 +91,22 @@ const services = {
...
+const setupApplicationMenu = async () => {
+  let status = ""
+  try {
+    status = await warnLatestVersion(version)
+  } catch (e) {
+    console.log('unable to verify latest version')
+    console.log(e)
+  }
+  applicationMenu(status === 'OUTDATED', (folder) => {
+    console.log('set folder', folder)
```
Comment (Aniket-Engg) [inline main.js:103]: Do you want to keep this console?

File: package.json (hunk 4 above)
Comment (Aniket-Engg) [reply package.json:74]: Would you not like to update the version in this PR?

File: main.js (hunk 1 above)
Comment (Aniket-Engg) [reply main.js]: ok

File: package.json (hunk 4 above)
Comment (yann300) [reply package.json:74]: yes I will once we've deployed the last remixd

Comment (yann300) [reply package.json:74]: (which is about to be published)

**Comment by yann300** [comment]
> Would you not like to update the version in this PR?

I 'll do that when remixd is released in a subsequent PR.

**Review by Aniket-Engg — APPROVED** [review]

**Pull request merged**